[Task] CreateRecording form does not recognize when targets reappear

If a target becomes briefly unreachable while filling out
the “Create recording” form, clicking the "Create" button will trigger
an expected HTTP 500 error. When the target becomes available again,
the Create button sometimes stays greyed out and the template dropdown
remains empty. Clicking the “refresh targets” button doesn’t update
the template dropdown. If you exit and re-enter the Create Recording
form then you can successfully create a recording.

Steps to reproduce:
1) Install cryostat operator in an Openshift cluster
2) Install a sample app to monitor:
```
oc new-app --docker-image=quay.io/andrewazores/quarkus-test:0.0.2
oc patch svc/quarkus-test -p '{"spec":{"$setElementOrder/ports":[{"port":9096},{"port":9999}],"ports":[{"name":"jfr-jmx","port":9096}]}}'
```
3) Select your target quarkus-test and click Create Recording
4) While viewing the form, restart the pod:
```
oc scale --replicas=0 deployment/quarkus-test
oc scale --replicas=1 deployment/quarkus-test
```
5) Notice the 'target appeared' notification was received. The templates dropdown should still be empty. Alternatively, if you filled out the entire Create Recording form before restarting the pod, the "Create" button might still be blue. Clicking on the blue button doesn't do anything.

```
Importing certificate /truststore/operator/..2022_08_04_20_03_47.2480200380/cryostat-sample-ca.crt ...
Certificate was added to keystore
[Storing /opt/cryostat.d/truststore.p12]
+ exec java -XX:+CrashOnOutOfMemoryError -Dcom.sun.management.jmxremote.port=9091 -Dcom.sun.management.jmxremote.rmi.port=9091 -Djavax.net.ssl.trustStore=/opt/cryostat.d/truststore.p12 -Djavax.net.ssl.trustStorePassword=WcqleLqoX09tpD8GM87khrU7zeFJV79V -Dcom.sun.management.jmxremote.autodiscovery=false -Dcom.sun.management.jmxremote.authenticate=true -Dcom.sun.management.jmxremote.password.file=/tmp/jmxremote.password -Dcom.sun.management.jmxremote.access.file=/tmp/jmxremote.access -Dcom.sun.management.jmxremote.ssl.need.client.auth=true -Djavax.net.ssl.keyStore=/var/run/secrets/operator.cryostat.io/cryostat-sample-tls/keystore.p12 -Djavax.net.ssl.keyStorePassword=7BR2SZMYIOky4Mtgg8pI -Dcom.sun.management.jmxremote.ssl=true -Dcom.sun.management.jmxremote.registry.ssl=true -cp '/app/resources:/app/classes:/app/libs/cryostat-core-2.12.0.jar:/app/libs/common-7.1.1.jar:/app/libs/encoder-1.2.2.jar:/app/libs/flightrecorder-7.1.1.jar:/app/libs/flightrecorder.rules-7.1.1.jar:/app/libs/flightrecorder.rules.jdk-7.1.1.jar:/app/libs/nashorn-core-15.4.jar:/app/libs/asm-7.3.1.jar:/app/libs/asm-commons-7.3.1.jar:/app/libs/asm-analysis-7.3.1.jar:/app/libs/asm-tree-7.3.1.jar:/app/libs/asm-util-7.3.1.jar:/app/libs/openshift-client-5.4.1.jar:/app/libs/openshift-model-5.4.1.jar:/app/libs/kubernetes-model-common-5.4.1.jar:/app/libs/jackson-annotations-2.11.2.jar:/app/libs/openshift-model-operator-5.4.1.jar:/app/libs/openshift-model-operatorhub-5.4.1.jar:/app/libs/openshift-model-monitoring-5.4.1.jar:/app/libs/openshift-model-console-5.4.1.jar:/app/libs/kubernetes-client-5.4.1.jar:/app/libs/kubernetes-model-core-5.4.1.jar:/app/libs/kubernetes-model-rbac-5.4.1.jar:/app/libs/kubernetes-model-admissionregistration-5.4.1.jar:/app/libs/kubernetes-model-apps-5.4.1.jar:/app/libs/kubernetes-model-autoscaling-5.4.1.jar:/app/libs/kubernetes-model-apiextensions-5.4.1.jar:/app/libs/kubernetes-model-batch-5.4.1.jar:/app/libs/kubernetes-model-certificates-5.4.1.jar:/app/libs/kubernetes-model-coordination-5.4.1.jar:/app/libs/kubernetes-model-discovery-5.4.1.jar:/app/libs/kubernetes-model-events-5.4.1.jar:/app/libs/kubernetes-model-extensions-5.4.1.jar:/app/libs/kubernetes-model-flowcontrol-5.4.1.jar:/app/libs/kubernetes-model-networking-5.4.1.jar:/app/libs/kubernetes-model-metrics-5.4.1.jar:/app/libs/kubernetes-model-policy-5.4.1.jar:/app/libs/kubernetes-model-scheduling-5.4.1.jar:/app/libs/kubernetes-model-storageclass-5.4.1.jar:/app/libs/kubernetes-model-node-5.4.1.jar:/app/libs/okhttp-3.12.12.jar:/app/libs/okio-1.15.0.jar:/app/libs/logging-interceptor-3.12.12.jar:/app/libs/slf4j-api-1.7.30.jar:/app/libs/jackson-dataformat-yaml-2.11.2.jar:/app/libs/snakeyaml-1.26.jar:/app/libs/jackson-datatype-jsr310-2.11.2.jar:/app/libs/jackson-databind-2.11.2.jar:/app/libs/jackson-core-2.11.2.jar:/app/libs/zjsonpatch-0.3.0.jar:/app/libs/generex-1.0.2.jar:/app/libs/automaton-1.11-8.jar:/app/libs/dagger-2.34.1.jar:/app/libs/javax.inject-1.jar:/app/libs/commons-lang3-3.12.0.jar:/app/libs/commons-codec-1.15.jar:/app/libs/commons-io-2.8.0.jar:/app/libs/commons-validator-1.7.jar:/app/libs/commons-beanutils-1.9.4.jar:/app/libs/commons-digester-2.1.jar:/app/libs/commons-logging-1.2.jar:/app/libs/commons-collections-3.2.2.jar:/app/libs/httpclient-4.5.13.jar:/app/libs/httpcore-4.4.13.jar:/app/libs/vertx-web-4.2.5.jar:/app/libs/vertx-web-common-4.2.5.jar:/app/libs/vertx-auth-common-4.2.5.jar:/app/libs/vertx-bridge-common-4.2.5.jar:/app/libs/vertx-core-4.2.5.jar:/app/libs/netty-common-4.1.74.Final.jar:/app/libs/netty-buffer-4.1.74.Final.jar:/app/libs/netty-transport-4.1.74.Final.jar:/app/libs/netty-handler-4.1.74.Final.jar:/app/libs/netty-codec-4.1.74.Final.jar:/app/libs/netty-tcnative-classes-2.0.48.Final.jar:/app/libs/netty-handler-proxy-4.1.74.Final.jar:/app/libs/netty-codec-socks-4.1.74.Final.jar:/app/libs/netty-codec-http-4.1.74.Final.jar:/app/libs/netty-codec-http2-4.1.74.Final.jar:/app/libs/netty-resolver-4.1.74.Final.jar:/app/libs/netty-resolver-dns-4.1.74.Final.jar:/app/libs/netty-codec-dns-4.1.74.Final.jar:/app/libs/vertx-web-client-4.2.5.jar:/app/libs/vertx-web-graphql-4.2.5.jar:/app/libs/graphql-java-17.3.jar:/app/libs/java-dataloader-3.1.0.jar:/app/libs/antlr4-runtime-4.9.2.jar:/app/libs/reactive-streams-1.0.3.jar:/app/libs/graphql-java-extended-scalars-17.0.jar:/app/libs/nimbus-jose-jwt-9.16.1.jar:/app/libs/jcip-annotations-1.0-1.jar:/app/libs/bcprov-jdk15on-1.69.jar:/app/libs/slf4j-jdk14-1.7.30.jar:/app/libs/gson-2.8.9.jar:/app/libs/caffeine-3.0.1.jar:/app/libs/jsoup-1.14.2.jar:/opt/cryostat.d/clientlib.d/*' @/app/jib-main-class-file
Aug 04, 2022 8:03:56 PM io.cryostat.core.log.Logger info
INFO: cryostat started.
Aug 04, 2022 8:03:56 PM io.cryostat.core.log.Logger info
INFO: Selected SSL KeyStore strategy with keystore /var/run/secrets/operator.cryostat.io/cryostat-sample-tls/keystore.p12
Aug 04, 2022 8:03:56 PM io.cryostat.core.log.Logger info
INFO: Local config path set as /opt/cryostat.d/conf.d
Aug 04, 2022 8:03:56 PM io.cryostat.core.log.Logger info
INFO: Selecting configured PlatformDetectionStrategy "io.cryostat.platform.internal.OpenShiftPlatformStrategy"
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: Selected OpenShift Platform Strategy
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: Selecting configured AuthManager "io.cryostat.net.openshift.OpenShiftAuthManager"
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: Local save path for flight recordings set as /opt/cryostat.d/recordings.d
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: HTTPS service running on https://cryostat-sample-myproject.apps-crc.testing:443
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: HTTP Server Verticle Started
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: Max concurrent WebSocket connections: 2147483647
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: MessagingServer Verticle Started
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: RuleProcessor Verticle Started
Aug 04, 2022 8:03:57 PM io.cryostat.core.log.Logger info
INFO: WebServer Verticle Started
Aug 04, 2022 8:04:07 PM io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure
WARNING: Exec Failure
java.net.SocketTimeoutException: Connect timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:546)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:133)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor.intercept(OpenShiftOAuthInterceptor.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

Aug 04, 2022 8:04:17 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:50268): GET /health/liveness 204 68ms
Aug 04, 2022 8:04:18 PM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1659643458},"message":{"event":{"kind":"LOST","serviceRef":{"connectUrl":"service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi","alias":"cryostat-sample-58c54d8966-pbvng","labels":{"app":"cryostat-sample","pod-template-hash":"58c54d8966","component":"cryostat","kind":"cryostat"},"annotations":{"platform":{"k8s.v1.cni.cncf.io/networks-status":"[{\n    \"name\": \"openshift-sdn\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.217.0.72\"\n    ],\n    \"default\": true,\n    \"dns\": {}\n}]","openshift.io/scc":"restricted","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"openshift-sdn\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.217.0.72\"\n    ],\n    \"default\": true,\n    \"dns\": {}\n}]"},"cryostat":{"HOST":"10.217.0.72","PORT":"9091","NAMESPACE":"myproject","POD_NAME":"cryostat-sample-58c54d8966-pbvng"}}}}}}
Aug 04, 2022 8:04:18 PM io.cryostat.core.log.Logger info
INFO: Outgoing WS message: {"meta":{"category":"TargetJvmDiscovery","type":{"type":"application","subType":"json"},"serverTime":1659643458},"message":{"event":{"kind":"FOUND","serviceRef":{"connectUrl":"service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi","alias":"cryostat-sample-58c54d8966-pbvng","labels":{"app":"cryostat-sample","pod-template-hash":"58c54d8966","component":"cryostat","kind":"cryostat"},"annotations":{"platform":{"k8s.v1.cni.cncf.io/networks-status":"[{\n    \"name\": \"openshift-sdn\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.217.0.72\"\n    ],\n    \"default\": true,\n    \"dns\": {}\n}]","openshift.io/scc":"restricted","k8s.v1.cni.cncf.io/network-status":"[{\n    \"name\": \"openshift-sdn\",\n    \"interface\": \"eth0\",\n    \"ips\": [\n        \"10.217.0.72\"\n    ],\n    \"default\": true,\n    \"dns\": {}\n}]"},"cryostat":{"HOST":"10.217.0.72","PORT":"9091","NAMESPACE":"myproject","POD_NAME":"cryostat-sample-58c54d8966-pbvng"}}}}}}
Aug 04, 2022 8:04:19 PM io.cryostat.core.log.Logger info
INFO: Creating connection for service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi
Aug 04, 2022 8:04:20 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48186): GET /api/v1/targets/10.217.0.72:9091/events 200 1138ms
Aug 04, 2022 8:04:20 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48186): GET /api/v1/targets/10.217.0.72:9091/templates 200 379ms
Aug 04, 2022 8:04:20 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48236): GET /api/v1/targets/10.217.0.72:9091/events 200 91ms
Aug 04, 2022 8:04:20 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48236): GET /api/v1/targets/10.217.0.72:9091/templates 200 102ms
Aug 04, 2022 8:04:20 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48244): GET /api/v1/targets/10.217.0.72:9091/events 200 62ms
Aug 04, 2022 8:04:21 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48244): GET /api/v1/targets/10.217.0.72:9091/templates 200 118ms
Aug 04, 2022 8:04:21 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48246): GET /api/v1/targets/10.217.0.72:9091/events 200 90ms
Aug 04, 2022 8:04:21 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:48246): GET /api/v1/targets/10.217.0.72:9091/templates 200 102ms
Aug 04, 2022 8:04:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:50512): GET /health/liveness 204 0ms
Aug 04, 2022 8:04:31 PM io.cryostat.core.log.Logger info
INFO: Removing cached connection for 10.217.0.72:9091: EXPIRED
Aug 04, 2022 8:04:31 PM io.cryostat.core.log.Logger info
INFO: Connection for service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi closed
Aug 04, 2022 8:04:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:50780): GET /health/liveness 204 0ms
Aug 04, 2022 8:04:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:51052): GET /health/liveness 204 0ms
Aug 04, 2022 8:04:56 PM io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure
WARNING: Exec Failure
java.io.EOFException
	at okio.RealBufferedSource.require(RealBufferedSource.java:61)
	at okio.RealBufferedSource.readByte(RealBufferedSource.java:74)
	at okhttp3.internal.ws.WebSocketReader.readHeader(WebSocketReader.java:117)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:101)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:274)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:214)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)

Aug 04, 2022 8:04:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:53812): GET /health/liveness 204 0ms
Aug 04, 2022 8:04:57 PM io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure
WARNING: Exec Failure
java.net.ConnectException: Failed to connect to /10.217.4.1:443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:249)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:133)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor.intercept(OpenShiftOAuthInterceptor.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:542)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	... 30 more

Aug 04, 2022 8:04:59 PM io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure
WARNING: Exec Failure
java.net.ConnectException: Failed to connect to /10.217.4.1:443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:249)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:133)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor.intercept(OpenShiftOAuthInterceptor.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:542)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	... 30 more

Aug 04, 2022 8:05:03 PM io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure
WARNING: Exec Failure
java.net.ConnectException: Failed to connect to /10.217.4.1:443
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:249)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:167)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:258)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:135)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:114)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:127)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.BackwardsCompatibilityInterceptor.intercept(BackwardsCompatibilityInterceptor.java:133)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.kubernetes.client.utils.ImpersonatorInterceptor.intercept(ImpersonatorInterceptor.java:68)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor.intercept(OpenShiftOAuthInterceptor.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:257)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:201)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.NioSocketImpl.timedFinishConnect(NioSocketImpl.java:542)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:597)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:633)
	at okhttp3.internal.platform.Platform.connectSocket(Platform.java:129)
	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:247)
	... 30 more

Aug 04, 2022 8:05:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:34126): GET /health/liveness 204 0ms
Aug 04, 2022 8:05:11 PM io.cryostat.core.log.Logger warn
WARNING: Exception thrown
io.fabric8.kubernetes.client.WatcherException: too old resource version: 36366 (36679)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$TypedWatcherWebSocketListener.onMessage(WatchConnectionManager.java:103)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:323)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:274)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:214)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: too old resource version: 36366 (36679)
	... 11 more

Aug 04, 2022 8:05:17 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:35476): GET /health/liveness 204 1ms
Aug 04, 2022 8:05:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:35884): GET /health/liveness 204 0ms
Aug 04, 2022 8:05:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:36124): GET /health/liveness 204 0ms
Aug 04, 2022 8:05:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:36358): GET /health/liveness 204 0ms
Aug 04, 2022 8:05:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:36612): GET /health/liveness 204 0ms
Aug 04, 2022 8:06:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:36844): GET /health/liveness 204 0ms
Aug 04, 2022 8:06:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:37086): GET /health/liveness 204 0ms
Aug 04, 2022 8:06:27 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:37330): GET /health/liveness 204 1ms
Aug 04, 2022 8:06:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:37754): GET /health/liveness 204 0ms
Aug 04, 2022 8:06:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:37978): GET /health/liveness 204 0ms
Aug 04, 2022 8:06:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:38226): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:38460): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:38670): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:38910): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:35 PM io.cryostat.core.log.Logger info
INFO: Creating connection for service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi
Aug 04, 2022 8:07:35 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:37000): GET /api/v1/targets/10.217.0.72:9091/events 200 219ms
Aug 04, 2022 8:07:35 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:37000): GET /api/v1/targets/10.217.0.72:9091/templates 200 26ms
Aug 04, 2022 8:07:35 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:37016): GET /api/v1/targets/10.217.0.72:9091/events 200 27ms
Aug 04, 2022 8:07:35 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.71:37016): GET /api/v1/targets/10.217.0.72:9091/templates 200 28ms
Aug 04, 2022 8:07:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:39144): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:46 PM io.cryostat.core.log.Logger info
INFO: Removing cached connection for 10.217.0.72:9091: EXPIRED
Aug 04, 2022 8:07:46 PM io.cryostat.core.log.Logger info
INFO: Connection for service:jmx:rmi:///jndi/rmi://10.217.0.72:9091/jmxrmi closed
Aug 04, 2022 8:07:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:39380): GET /health/liveness 204 0ms
Aug 04, 2022 8:07:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:39686): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:39902): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:40118): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:40344): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:40562): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:40792): GET /health/liveness 204 0ms
Aug 04, 2022 8:08:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:41044): GET /health/liveness 204 0ms
Aug 04, 2022 8:09:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:41264): GET /health/liveness 204 0ms
Aug 04, 2022 8:09:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:41476): GET /health/liveness 204 0ms
Aug 04, 2022 8:09:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:41708): GET /health/liveness 204 0ms
Aug 04, 2022 8:09:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:41930): GET /health/liveness 204 0ms
Aug 04, 2022 8:09:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:42148): GET /health/liveness 204 1ms
Aug 04, 2022 8:09:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:42396): GET /health/liveness 204 0ms
Aug 04, 2022 8:10:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:42618): GET /health/liveness 204 0ms
Aug 04, 2022 8:10:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:42860): GET /health/liveness 204 0ms
Aug 04, 2022 8:10:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:43090): GET /health/liveness 204 0ms
Aug 04, 2022 8:10:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:43306): GET /health/liveness 204 0ms
Aug 04, 2022 8:10:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:43526): GET /health/liveness 204 1ms
Aug 04, 2022 8:10:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:43832): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:44052): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:44258): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:26 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:44488): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:36 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:44704): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:46 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:44916): GET /health/liveness 204 0ms
Aug 04, 2022 8:11:56 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:45164): GET /health/liveness 204 0ms
Aug 04, 2022 8:12:06 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:45384): GET /health/liveness 204 0ms
Aug 04, 2022 8:12:16 PM io.cryostat.core.log.Logger info
INFO: (10.217.0.1:45594): GET /health/liveness 204 0ms
```

Here's a recent log of 2.2.0-dev in crc.

Cryostat initially had a working watcher connection and soon after startup it even detected itself coming online and logged its own WebSocket notifications about that (no client was connected to actually send those messages to, but it logs regardless). Then we see some repeated `io.fabric8.kubernetes.client.dsl.internal.WatcherWebSocketListener onFailure` occurrences. There is an initial one at `Aug 04, 2022 8:04:07 PM`. The next at `Aug 04, 2022 8:04:56 PM`, then `Aug 04, 2022 8:04:57 PM` (+1s), `Aug 04, 2022 8:04:59 PM` (+2s), and `Aug 04, 2022 8:05:03 PM` (+4s). This looks like the exponential backoff interval logic that I found in the fabric8 sources.

Then we get:

```
Aug 04, 2022 8:05:11 PM io.cryostat.core.log.Logger warn
WARNING: Exception thrown
io.fabric8.kubernetes.client.WatcherException: too old resource version: 36366 (36679)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager$TypedWatcherWebSocketListener.onMessage(WatchConnectionManager.java:103)
	at okhttp3.internal.ws.RealWebSocket.onReadMessage(RealWebSocket.java:323)
	at okhttp3.internal.ws.WebSocketReader.readMessageFrame(WebSocketReader.java:219)
	at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.java:105)
	at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.java:274)
	at okhttp3.internal.ws.RealWebSocket$2.onResponse(RealWebSocket.java:214)
	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:203)
	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: io.fabric8.kubernetes.client.KubernetesClientException: too old resource version: 36366 (36679)
	... 11 more
```

after this there are no more logged exceptions that look like watcher reconnection attempts.

We can (and do) catch these exceptions, but currently they are only logged and nothing else happens. The `WatcherException` in kubernetes-client 5.4.1 that we are using has a method `isShouldRetry()`. I will try logging that as well and reproduce the failure. I expect that we will see `true` on the initial exceptions and `false` on the last one. If so, then if we get an exception with the `isShouldRetry() == false`, we can close the watcher object and attempt to re-open a new one. Hopefully this doesn't just immediately run into the same `WatcherException: too old resource version: 36366 (36679)`.

--- a/src/app/Shared/MatchExpressionEvaluator.tsx
+++ b/src/app/Shared/MatchExpressionEvaluator.tsx
@@ -60,7 +60,7 @@ export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEv
     addSubscription(
       context.target.target().subscribe(setTarget)
     );
-  }, [context, context.target]);
+  }, [addSubscription, context, context.target]);
 
   React.useEffect(() => {
     if (!props.matchExpression || !target?.connectUrl) {

--- a/src/app/TargetSelect/TargetSelect.tsx
+++ b/src/app/TargetSelect/TargetSelect.tsx
@@ -148,7 +148,7 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props) 
       }
     }
     setExpanded(false);
-  }, [context, context.target, selected, notifications, setExpanded, setCachedTargetSelection, removeCachedTargetSelection]);
+  }, [context, context.target, selected, notifications, setExpanded]);
 
   const selectNone = React.useCallback(() => {
     onSelect(undefined, undefined, true);

--- a/src/app/TargetSelect/TargetSelect.tsx
+++ b/src/app/TargetSelect/TargetSelect.tsx
@@ -47,6 +47,7 @@ import { catchError, first } from 'rxjs/operators';
 
 import { CreateTargetModal } from './CreateTargetModal';
 import _ from 'lodash';
+import { NotificationCategory } from '@app/Shared/Services/NotificationChannel.service';
 
 export interface TargetSelectProps {
 }
@@ -63,12 +64,33 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props) 
   const addSubscription = useSubscriptions();
 
   React.useEffect(() => {
-    const sub = context.targets.targets().subscribe((targets) => {
-      setTargets(targets);
-      selectTargetFromCache(targets);
-    });
-    return () => sub.unsubscribe();
+    addSubscription(
+      context.targets.targets().subscribe((targets) => {
+        setTargets(targets);
+        selectTargetFromCache(targets);
+      })
+    );
   }, [context, context.targets, setTargets]);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.notificationChannel.messages(NotificationCategory.TargetJvmDiscovery)
+        .subscribe(v => {
+          const evt: TargetDiscoveryEvent = v.message.event;
+          if (evt.kind === 'LOST') {
+            const target: Target = {
+              connectUrl: evt.serviceRef.connectUrl,
+              alias: evt.serviceRef.alias,
+            }
+            context.target.target().subscribe((currentTarget) => {
+              if (currentTarget.connectUrl == target.connectUrl && currentTarget.alias == target.alias) {
+                selectNone();
+              }
+            })
+          }
+        })
+    );
+  }, [addSubscription, context, context.notificationChannel, context.target]);
 
   const refreshTargetList = React.useCallback(() => {
     setLoading(true);
@@ -126,7 +148,7 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props) 
       }
     }
     setExpanded(false);
-  }, [context, selected, notifications, setExpanded]);
+  }, [context, context.target, selected, notifications, setExpanded, setCachedTargetSelection, removeCachedTargetSelection]);
 
   const selectNone = React.useCallback(() => {
     onSelect(undefined, undefined, true);

--- a/src/app/TargetSelect/TargetSelect.tsx
+++ b/src/app/TargetSelect/TargetSelect.tsx
@@ -82,8 +82,8 @@ export const TargetSelect: React.FunctionComponent<TargetSelectProps> = (props) 
               connectUrl: evt.serviceRef.connectUrl,
               alias: evt.serviceRef.alias,
             }
-            context.target.target().subscribe((currentTarget) => {
-              if (currentTarget.connectUrl == target.connectUrl && currentTarget.alias == target.alias) {
+            context.target.target().pipe(first()).subscribe((currentTarget) => {
+              if (currentTarget.connectUrl === target.connectUrl && currentTarget.alias === target.alias) {
                 selectNone();
               }
             })

--- a/src/test/Rules/CreateRule.test.tsx
+++ b/src/test/Rules/CreateRule.test.tsx
@@ -48,6 +48,7 @@ import { ServiceContext, defaultServices } from '@app/Shared/Services/Services';
 import { CreateRule } from '@app/Rules/CreateRule';
 import { EventTemplate } from '@app/CreateRecording/CreateRecording';
 import { Target } from '@app/Shared/Services/Target.service';
+import { NotificationMessage } from '@app/Shared/Services/NotificationChannel.service';
 
 const escapeKeyboardInput = (value: string) => {
   return value.replace(/[{[]/g, '$&$&');
@@ -82,6 +83,18 @@ const mockRule: Rule =  {
   maxSizeBytes: 0
 };
 
+const mockNewTarget: Target = {
+  connectUrl: 'service:jmx:rmi://someUrl1',
+  alias: 'someAlias',
+}
+
+const mockTargetFoundNotification = {
+  message: {
+    event: { kind: 'FOUND', serviceRef: mockNewTarget }
+  }
+} as NotificationMessage;
+
+
 const history = createMemoryHistory();
 
 jest.mock('react-router-dom', () => ({
@@ -91,6 +104,7 @@ jest.mock('react-router-dom', () => ({
 }));
 
 const createSpy = jest.spyOn(defaultServices.api, 'createRule').mockReturnValue(of(true));
+jest.spyOn(defaultServices.notificationChannel, 'messages').mockReturnValue(of(mockTargetFoundNotification));
 jest.spyOn(defaultServices.api, 'doGet').mockReturnValue(of([mockEventTemplate]));
 jest.spyOn(defaultServices.target, 'target').mockReturnValue(of(mockTarget));
 jest.spyOn(defaultServices.targets, 'targets').mockReturnValue(of([mockTarget]));

